Sources/PrimerSDK/Classes/Core/PrimerHeadlessUniversalCheckout/Composable/NolPay/NolPayPhoneMetadataService.swift

```diff
@@ -0,0 +1,65 @@
+//
+//  NolPayPhoneMetadataService.swift
+//  PrimerSDK
+//
+//  Created by Boris on 27.10.23..
+//
+
+import Foundation
+
+typealias PhoneMetadataCompletion = (Result<(PrimerValidationStatus, String?, String?), PrimerError>) -> Void
+
+protocol NolPayPhoneMetadataProviding {
+    func getPhoneMetadata(mobileNumber: String, completion: @escaping PhoneMetadataCompletion)
+}
+
+struct NolPayPhoneMetadataService: NolPayPhoneMetadataProviding {
+    func getPhoneMetadata(mobileNumber: String, completion: @escaping PhoneMetadataCompletion) {
+        
+        guard let clientToken = PrimerAPIConfigurationModule.decodedJWTToken else {
+            let err = PrimerError.invalidClientToken(userInfo: ["file": #file,
+                                                                "class": "\(Self.self)",
+                                                                "function": #function,
+                                                                "line": "\(#line)"],
+                                                     diagnosticsId: UUID().uuidString)
+            ErrorHandler.handle(error: err)
+            completion(.failure(err))
+            return
+        }
+
+        let requestBody = Request.Body.PhoneMetadata.PhoneMetadataDataRequest(phoneNumber: mobileNumber)
+        let client = PrimerAPIClient()
+        client.getPhoneMetadata(clientToken: clientToken, paymentRequestBody: requestBody) { result in
+
+            switch result {
+            case .success(let phoneMedatadaResponse):
+                let countryCode = phoneMedatadaResponse.countryCode
+                let mobileNumber = phoneMedatadaResponse.nationalNumber
```

Comment:
```suggestion
            case .success(let phoneMetadataResponse):
                let countryCode = phoneMetadataResponse.countryCode
                let mobileNumber = phoneMetadataResponse.nationalNumber
```

Reply:
Whats this? @jnewc

Reply:
Good catch...

me**d**ada**t**aResponse -> me**t**a**d**ataResponse

Reply:
ah! 🤦 thanks! hahahah